pulpcore/app/viewsets/repository.py

```diff
@@ -272,6 +278,78 @@ class RemoteViewSet(
     queryset = Remote.objects.all()
     filterset_class = RemoteFilter
 
+    @extend_schema(
+        description="Trigger an asynchronous delete task",
+        responses={202: AsyncOperationResponseSerializer},
+    )
+    def destroy(self, request, pk, **kwargs):
+        """Remove the remote only if there is no remote artifact using it."""
+        versions = _get_versions_using_remote(pk)
+        if not versions:
+            return super().destroy(request, pk, **kwargs)
+
+        repos_with_del_permissions = self.get_repositories_with_delete_permission(versions)
+
+        if repos_with_del_permissions.exists():
+            versions_urls = []
+            for version in versions:
+                if version.repository.pk in repos_with_del_permissions:
+                    version_url = urljoin(
+                        get_url(version.repository), f"versions/{version.number}/"
+                    )
+                    versions_urls.append(version_url)
+
+            return Response(
+                _(
+                    "Attempt to delete the remote which is still used and referenced from the "
+                    "following repository versions: {}. Try again after removing the listed "
+                    "repository versions and associated remote (content) artifacts."
```

Comment:
```suggestion
                    "Attempt to delete the remote which is still used and referenced from the "
                    "following repository versions below. Try again after removing the listed "
                    "repository versions and associated remote (content) artifacts.\n\n{}"
```

A sentence on top and bottom of a potentially large list would be hard to read (to me).